core/field_colour_slider.js

```diff
@@ -0,0 +1,346 @@
+  var thisField = this;
+  Blockly.FieldColourSlider.activateEyedropper_(function(value) {
+    thisField.setValue(value);
+  });
+};
+
+/**
+ * Create a palette under the colour field.
+ * @private
+ */
+Blockly.FieldColourSlider.prototype.showEditor_ = function() {
+  Blockly.DropDownDiv.hideWithoutAnimation();
+  Blockly.DropDownDiv.clearContent();
+  var div = Blockly.DropDownDiv.getContentDiv();
+
+  var hueElements = this.createLabelDom_('Hue');
+  div.appendChild(hueElements[0]);
+  this.hueReadout_ = hueElements[1];
+  this.hueSlider_ = new goog.ui.Slider();
+  this.hueSlider_.setUnitIncrement(5);
+  this.hueSlider_.setMinimum(0);
+  this.hueSlider_.setMaximum(359);
+  this.hueSlider_.render(div);
+
+  var saturationElements = this.createLabelDom_('Saturation');
+  div.appendChild(saturationElements[0]);
+  this.saturationReadout_ = saturationElements[1];
+  this.saturationSlider_ = new goog.ui.Slider();
+  this.saturationSlider_.setUnitIncrement(0.01);
+  this.saturationSlider_.setStep(0.001);
+  this.saturationSlider_.setMinimum(0.01);
+  this.saturationSlider_.setMaximum(0.99);
+  this.saturationSlider_.render(div);
+
+  var brightnessElements = this.createLabelDom_('Brightness');
+  div.appendChild(brightnessElements[0]);
+  this.brightnessReadout_ = brightnessElements[1];
+  this.brightnessSlider_ = new goog.ui.Slider();
+  this.brightnessSlider_.setUnitIncrement(2);
+  this.brightnessSlider_.setMinimum(5);
+  this.brightnessSlider_.setMaximum(255);
+  this.brightnessSlider_.render(div);
+
+  Blockly.FieldColourSlider.hueChangeEventKey_ = goog.events.listen(this.hueSlider_,
+        goog.ui.Component.EventType.CHANGE,
+        this.sliderCallbackFactory('hue'));
+  Blockly.FieldColourSlider.saturationChangeEventKey_ = goog.events.listen(this.saturationSlider_,
+        goog.ui.Component.EventType.CHANGE,
+        this.sliderCallbackFactory('saturation'));
+  Blockly.FieldColourSlider.brightnessChangeEventKey_ = goog.events.listen(this.brightnessSlider_,
+        goog.ui.Component.EventType.CHANGE,
+        this.sliderCallbackFactory('brightness'));
+
+  if (Blockly.FieldColourSlider.activateEyedropper_) {
+    var button = document.createElement('button');
+    button.setAttribute('class', 'scratchEyedropper');
+    var image = document.createElement('img');
+    image.src = Blockly.mainWorkspace.options.pathToMedia + Blockly.FieldColourSlider.EYEDROPPER_PATH;
+    button.appendChild(image);
+    div.appendChild(button);
+    Blockly.FieldColourSlider.eyedropperEventData_ = Blockly.bindEventWithChecks_(button,
+      'mousedown',
+      this,
+      this.activateEyedropperInternal_
+    );
+  }
+
+  Blockly.DropDownDiv.setColour('#ffffff', '#dddddd');
+  Blockly.DropDownDiv.setCategory(this.sourceBlock_.parentBlock_.getCategory());
+  Blockly.DropDownDiv.showPositionedByBlock(this, this.sourceBlock_);
+
+  this.setValue(this.getValue());
+};
+
+Blockly.FieldColourSlider.prototype.dispose = function() {
```

Comment:
do you need the widgetdiv hideIfOwner call?

Reply:
Oh it doesn't use the widgetdiv anymore (uses the dropdowndiv)